staging/vhost-device-gpu/src/main.rs

```diff
@@ -0,0 +1,140 @@
+// VIRTIO GPU Emulation via vhost-user
+//
+// Copyright 2024 Red Hat Inc
+//
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+use log::{error, info};
+use std::{path::PathBuf, process::exit};
+
+use clap::{Parser, ValueEnum};
+use thiserror::Error as ThisError;
+use vhost_device_gpu::{
+    device::{self, VhostUserGpuBackend},
+    GpuConfig, GpuMode,
+};
+use vhost_user_backend::VhostUserDaemon;
+use vm_memory::{GuestMemoryAtomic, GuestMemoryMmap};
+
+type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug, ThisError)]
+pub(crate) enum Error {
+    #[error("Could not create backend: {0}")]
+    CouldNotCreateBackend(device::Error),
+    #[error("Could not create daemon: {0}")]
+    CouldNotCreateDaemon(vhost_user_backend::Error),
+    #[error("Fatal error: {0}")]
+    ServeFailed(vhost_user_backend::Error),
+}
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct GpuArgs {
+    /// vhost-user Unix domain socket.
+    #[clap(short, long, value_name = "SOCKET")]
+    socket_path: PathBuf,
+    #[clap(short, long, value_enum)]
+    renderer: RenderMode,
+}
+
+#[derive(Debug, Clone, ValueEnum)]
+enum RenderMode {
+    Gfxstream,
+    Virglrenderer,
+}
+
+impl From<RenderMode> for GpuMode {
+    fn from(mode: RenderMode) -> Self {
+        match mode {
+            RenderMode::Gfxstream => GpuMode::ModeGfxstream,
+            RenderMode::Virglrenderer => GpuMode::ModeVirglRenderer,
+        }
+    }
+}
+
+impl TryFrom<GpuArgs> for GpuConfig {
+    type Error = Error;
+
+    fn try_from(args: GpuArgs) -> Result<Self> {
+        let socket_path = args.socket_path;
+        let renderer: GpuMode = args.renderer.into();
+
+        Ok(GpuConfig::new(socket_path, renderer))
+    }
+}
+
+fn start_backend(config: GpuConfig) -> Result<()> {
+    info!("Starting backend");
+    let socket = config.get_socket_path();
+    let backend = VhostUserGpuBackend::new(config).map_err(Error::CouldNotCreateBackend)?;
+
+    let mut daemon = VhostUserDaemon::new(
+        "vhost-device-gpu-backend".to_string(),
+        backend.clone(),
+        GuestMemoryAtomic::new(GuestMemoryMmap::new()),
+    )
+    .map_err(Error::CouldNotCreateDaemon)?;
+
+    backend.set_epoll_handler(&daemon.get_epoll_handlers());
+
+    daemon.serve(socket).map_err(Error::ServeFailed)?;
+    Ok(())
+}
+
+#[cfg(not(target_env = "musl"))]
+fn main() {
+    env_logger::init();
+
+    if let Err(e) = start_backend(GpuConfig::try_from(GpuArgs::parse()).unwrap()) {
+        error!("{e}");
+        exit(1);
+    }
+}
+
+// Rust vmm container (https://github.com/rust-vmm/rust-vmm-container) doesn't
+// have tools to do a musl build at the moment, and adding that support is
+// tricky as well to the container. Skip musl builds until the time pre-built
+// rutabaga library is available for musl.
+#[cfg(target_env = "musl")]
+fn main() {}
```

Comment:
You can do something like 
```
#[cfg(target_env = "musl")]
compile_error!("musl is not supported yet");
```


and you won't have to guard the main fn as the binary won't compile at all for musl.